core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -867,43 +867,98 @@ def _compute_permutation_importance(self, X, y, features: list, eval_metric=None
+        if not self.is_fit():
+            return False
+        # Get model specific compiler options
+        # Model type can be described with either model type, or model name as string
+        if type(self) in compiler_configs:
+            configs = compiler_configs[type(self)]
+        elif self.name in compiler_configs:
+            configs = compiler_configs[self.name]
+        else:
+            configs = compiler_configs
+        compiler = configs.get("compiler", "native")
+        compiler_fallback_to_native = configs.get('compiler_fallback_to_native', False)
+
+        compilers = self._valid_compilers()
+        compiler_names = {c.name: c for c in compilers}
+        if compiler is not None and compiler not in compiler_names:
+            return False
+        compiler_cls = compiler_names[compiler]
+        if not compiler_cls.can_compile():
+            if not compiler_fallback_to_native:
+                return False
+        return True
+
+    def compile(self, compiler_configs=None):
+        """
+        Compile the trained model for faster inference.
+
+        NOTE:
+        - The model is assumed to be fitted before compilation.
+        - If save_in_pkl attribute of the compiler is False, self.model would be set to None.
+          The compiled model can be loaded via load_model method.
+
+        Parameters
+        ----------
+        compiler_configs : dict, default=None
+            Model specific compiler options.
+            This can be useful to specify the compiler backend for a specific model,
+            e.g. {"RandomForest": {"compiler": "onnx"}}
+        """
+        assert self.is_fit(), "The model must be fit before calling the compile method."
+        if compiler_configs is None:
+            compiler_configs = {}
+
+        # Get model specific compiler options
+        # Model type can be described with either model type, or model name as string
+        if type(self) in compiler_configs:
+            configs = compiler_configs[type(self)]
+        elif self.name in compiler_configs:
+            configs = compiler_configs[self.name]
+        else:
+            configs = compiler_configs
+        compiler = configs.get("compiler", "native")
+        batch_size = configs.get("batch_size", None)
+        compiler_fallback_to_native = configs.get('compiler_fallback_to_native', False)
+
+        self._compiler = self._get_compiler(compiler=compiler,
+                                            compiler_fallback_to_native=compiler_fallback_to_native)
+        if self._compiler is not None:
+            input_types = self._get_input_types(batch_size=batch_size)
+            self._compiler.compile(model=self.model, path=self.path, input_types=input_types)
+            if not self._compiler.save_in_pkl:
+                self.model = None
+            self.save(path=self.path, verbose=False)
+
+    def _get_input_types(self, batch_size=None):
+        return [((batch_size, len(self._features)), np.float32)]
```

Comment:
This is a false assumption. I guess we can wait to fix it until it becomes a problem, but `self._features` is not the truth of the input shape.

Reply:
Need to add documentation to this method

Reply:
Please add `# FIXME: This won't work for all models, and self._features is not a trustworthy variable for final input shape`

Reply:
> This is a false assumption. 

Yes, this may not be true for all models. I think we can override this method when it's not applicable.

Added document and the comment to keep further notice.